include/SongEditor.h

```diff
@@ -45,15 +46,44 @@ class Song;
 class TextFloat;
 class TimeLineWidget;
 
+Q_DECLARE_METATYPE(QLinearGradient)
+
 class positionLine : public QWidget
 {
+	Q_OBJECT
+	Q_PROPERTY ( bool tailGradient READ tailGradient WRITE setTailGradient )
+	Q_PROPERTY ( QColor lineColor READ lineColor WRITE setLineColor )
 public:
-	positionLine( QWidget * parent );
+	positionLine ( QWidget * parent, ComboBoxModel * zoom, Song * song );
+	
+	int width ();
+	void go ( int x, int y );	// NOTE: Check SongEditor.cpp for usage
+	void zoomUpdate ();
+	void playStateChanged ();
+	
+	// qproperty access functions
+	bool tailGradient () const;
+	void setTailGradient ( const bool & g );
+	QColor lineColor () const;
+	void setLineColor ( const QColor & c );
 
 private:
 	void paintEvent( QPaintEvent * pe ) override;
+	
+	int m_width;
+	int m_x, m_y;	// NOTE: m_x is the playback position, not the position where the tail starts
```

Comment:
Is it possible to get rid of `m_width`, `m_x`, and `m_y` and just use the position and size properties inherited from `QWidget`? Redundant information is just a place for bugs to hide should it ever get out of sync, and the code is easier to read if it sticks to standard methods.

Reply:
It's possible, I've figured out a way to get around without them. See commit.